frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-form/upload-pvc-form.tsx

```diff
@@ -0,0 +1,341 @@
+  const [error, setError] = React.useState('');
+  const [isAllocating, setIsAllocating] = React.useState(false);
+  const [dvObj, setDvObj] = React.useState<V1alpha1DataVolume>(null);
+
+  const { uploads, uploadData } = React.useContext(CDIUploadContext);
+  const { namespace } = props;
+  const title = 'Upload Data to Persistent Volume Claim';
+
+  const save = (e: React.FormEvent<EventTarget>) => {
+    e.preventDefault();
+    if (!fileName) {
+      setError('File input is missing');
+    } else {
+      setError('');
+      setIsAllocating(true);
+      setIsSubmitting(true);
+      // eslint-disable-next-line promise/catch-or-return
+      createUploadPVC(dvObj).then(
+        ({ token }) => {
+          setIsAllocating(false);
+          uploadData({
+            file: fileValue,
+            token,
+            pvcName: getName(dvObj),
+            namespace,
+          });
+        },
+        ({ message }: { message: string }) => {
+          setIsAllocating(false);
+          setError(message || 'Could not create persistent volume claim.');
+        },
+      );
+    }
+  };
+
+  const handleFileChange = (value, filename) => {
+    setFileName(filename);
+    setFileValue(value);
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <div
+        className={cx('co-m-pane__body co-m-pane__form', {
+          'kv--create-upload__hide': isSubmitting,
+        })}
+      >
+        <h1 className="co-m-pane__heading co-m-pane__heading--baseline">
+          <div className="co-m-pane__name">{title}</div>
+        </h1>
+        <form className="co-m-pane__body-group" onSubmit={save}>
+          <UploadPVCForm
+            onChange={setDvObj}
+            namespace={namespace}
+            fileValue={fileValue}
+            fileName={fileName}
+            handleFileChange={handleFileChange}
+          />
+          <ButtonBar errorMessage={error}>
+            <ActionGroup className="pf-c-form">
+              <Button id="save-changes" type="submit" variant="primary">
+                Upload
+              </Button>
+              <Button onClick={history.goBack} type="button" variant="secondary">
+                Cancel
+              </Button>
+            </ActionGroup>
+          </ButtonBar>
+        </form>
+      </div>
+      <UploadPVCFormStatus
+        upload={uploads.find(
+          (upl) => upl.pvcName === getName(dvObj) && upl.namespace === namespace,
+        )}
+        isSubmitting={isSubmitting}
+        isAllocating={isAllocating}
+        allocateError={error}
+        onErrorClick={() => {
+          setIsSubmitting(false);
+          setError('');
+        }}
+        onSuccessClick={() =>
+          history.push(resourcePath(PersistentVolumeClaimModel.kind, getName(dvObj), namespace))
+        }
+        onCancelFinish={() => history.push(resourcePath(PersistentVolumeClaimModel.kind))}
+      />
+    </>
+  );
+};
+
+export const UploadPVC = (props) => {
```

Comment:
i dont really see a need for this component, just move the `params` logic to `UploadPVCPage`